pallets/staking_rewards/src/lib.rs

```diff
@@ -0,0 +1,235 @@
+            reward_curve,
+            total_staked,
+            total_issuance,
+        );
+        let reward_proportion_of_max =
+            Perbill::from_rational(reward_as_per_npos, reward_curve.maximum * total_issuance);
+        reward_proportion_of_max
+    }
+
+    /// Get emission per year according to NPoS as described in token economics doc here
+    /// https://research.web3.foundation/en/latest/polkadot/overview/2-token-economics.html. Doesn't
+    /// consider remaining emission supply
+    fn get_yearly_emission_reward_as_per_npos_only(
+        reward_curve: &PiecewiseLinear,
+        total_staked: BalanceOf<T>,
+        total_issuance: BalanceOf<T>,
+    ) -> BalanceOf<T> {
+        reward_curve.calculate_for_fraction_times_denominator(total_staked, total_issuance)
+    }
+
+    // TODO: Needed as RPC?
+    /// Get maximum emission per year according to the decay percentage and given emission supply
+    fn get_max_yearly_emission(emission_supply: BalanceOf<T>) -> BalanceOf<T> {
+        // Emission supply decreases by "decay percentage" of the remaining emission supply per year
+        Percent::from_percent(T::RewardDecayPct::get()) * emission_supply
+    }
+
+    /// Given yearly emission rewards, calculate for an era.
+    fn get_emission_reward_for_era_given_yearly(
+        era_duration_millis: u64,
+        yearly_rewards: BalanceOf<T>,
+    ) -> BalanceOf<T> {
+        // Ratio of milliseconds in an era to milliseconds in an year
+        let portion = Perbill::from_rational(era_duration_millis, MILLISECONDS_PER_YEAR);
+        portion * yearly_rewards
+    }
+
+    /// Set emission supply. Used to set the reduced supply after emitting rewards
+    fn set_new_emission_supply(supply: BalanceOf<T>) {
+        <StakingEmissionSupply<T>>::put(supply)
+    }
+
+    /// If emission supply for staking is 0 (on starting PoS), set it to the remaining emission
+    /// supply from PoA and reset it to making this function idempotent unless emission supply in
+    ///PoA module is set again
+    fn set_emission_supply_from_poa() -> Weight {
+        if Self::staking_emission_supply().is_zero() {
+            let supply = <poa::EmissionSupply<T>>::take();
+            Self::set_new_emission_supply(supply);
+            T::DbWeight::get().reads_writes(2, 2)
+        } else {
+            T::DbWeight::get().reads(1)
+        }
+    }
+}
+
+impl<T: Config> EraPayout<BalanceOf<T>> for Module<T> {
+    /// Compute era payout for validators and treasury and reduce the remaining emission supply.
+    /// It is assumed and expected that this is called only when a payout of an era has to computed
+    /// and isn't called twice for the same era as it has a side-effect (reducing remaining supply).
+    /// Currently it doesn't seem possible to avoid this side effect as there is no way for this pallet
+    /// to be notified if an era payout was successfully done.
```

Comment:
would be nice to repeat here in the comment what the two return values are (stakers reward  / treasury rewards)
(even if it can be guessed from the code or looked up in the EraPayout docs)